File: src/main/java/jadx/plugins/example/AliucordJADXOptions.java
package jadx.plugins.example;

import jadx.api.plugins.options.impl.BasePluginOptionsBuilder;

public class AliucordJADXOptions extends BasePluginOptionsBuilder {

	private CodegenLanguage codegenLanguage;
	private boolean useFullName;

	@Override
	public void registerOptions() {
		enumOption(AliucordJADX.PLUGIN_ID + ".codegen", CodegenLanguage.values(), CodegenLanguage::valueOf)
				.description("snippet code language")
				.defaultValue(CodegenLanguage.KOTLIN)
				.setter(v -> codegenLanguage = v);

		boolOption(AliucordJADX.PLUGIN_ID + ".fullName")
				.description("use full name for identifiers")
				.defaultValue(true)
				.setter(v -> useFullName = v);
	}

	public CodegenLanguage getCodegenLanguage() {
		return codegenLanguage;
	}
	public boolean useFullName() { return useFullName; }
}

File: src/main/java/com/github/canny1913/AliucordJADXOptions.kt
package com.github.canny1913

import jadx.api.plugins.options.impl.BasePluginOptionsBuilder

class AliucordJADXOptions: BasePluginOptionsBuilder() {
	var useFullName: Boolean = true
		private set
	lateinit var codegenLanguage: CodegenLanguage
		private set

	override fun registerOptions() {
		enumOption(AliucordJADX.PLUGIN_ID + ".codegen", CodegenLanguage.entries.toTypedArray(), CodegenLanguage::valueOf)
			.description("snippet code language")
			.defaultValue(CodegenLanguage.KOTLIN)
			.setter { codegenLanguage = it }

		boolOption(AliucordJADX.PLUGIN_ID + ".fullName")
			.description("use full name for identifiers")
			.defaultValue(true)
			.setter { useFullName = it }
	}
}
